### PR TITLE
docs(react-swatch-picker): added closing of popup on color select

### PR DESCRIPTION
**Pull request by ValentinaKozlova**
## Previous Behavior
- Popup was opened when color was selected

## New Behavior
- Popup closes on color select
- Added more information to BestPractices

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx [hunk 1]
```diff
@@ -48,29 +60,43 @@ const gradientColors = [
...
   return (
     <>
-      <Popover trapFocus>
-        <PopoverTrigger disableButtonEnhancement>
-          <Button>Choose color</Button>
-        </PopoverTrigger>
-
-        <PopoverSurface>
-          <h3>Color set 1</h3>
+      <Button {...restoreFocusTargetAttribute} ref={buttonRef} onClick={() => setOpen(_open => !_open)}>
```
[the remark this answers is not in the document]
Comment (layershifter) [reply packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx:85]: From what I see, it's enough to make it controlled, https://stackblitz.com/edit/wqakma?file=src%2Fexample.tsx

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+import {
+  makeStyles,
+  shorthands,
+  Button,
+  Popover,
+  PopoverSurface,
+  useRestoreFocusTarget,
+  useId,
+} from '@fluentui/react-components';
 import { SwatchPicker, SwatchPickerOnSelectEventHandler, ColorSwatch } from '@fluentui/react-swatch-picker-preview';
```
Comment (marcosmoura) [inline packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx:11]: As we are importing values and types separately from `@fluentui/react-components`, can we do the same with `@fluentui/react-swatch-picker-preview`?

Comment (ValentinaKozlova) [reply packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx:11]: we don't have it in `react-components` yet

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ValentinaKozlova) [reply packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx:85]: You're right, it works

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerBestPractices.md [hunk 3]
```diff
@@ -8,11 +8,13 @@
 - Maximum recommended number of swatches per row is 8.
```
Comment (fabricteam) [inline packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerBestPractices.md:8]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31360&commitId=79c05e363bdc5f40c267234ce476a9a849a0b222&env=prod&clientType=FLUENTUI)


  <details><summary><b>SwatchPicker Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.Shape.default.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx (hunk 2 above)
Comment (marcosmoura) [inline packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx]: Sorry, what I meant was from the `@fluentui/react-swatch-picker-preview`.

```suggestion
import { SwatchPicker, ColorSwatch } from '@fluentui/react-swatch-picker-preview';
import type { SwatchPickerOnSelectEventHandler } from '@fluentui/react-swatch-picker-preview';
```

**Review by marcosmoura — APPROVED** [review]

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx (hunk 2 above)
Comment (ValentinaKozlova) [reply packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerPopup.stories.tsx]: fixed :)

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**